clang/include/clang/CAS/IncludeTree.h

```diff
@@ -67,19 +69,52 @@ class IncludeTree : public IncludeTreeBase<IncludeTree> {
   Expected<FileInfo> getBaseFileInfo();
 
   SrcMgr::CharacteristicKind getFileCharacteristic() const {
-    return (SrcMgr::CharacteristicKind)dataSkippingIncludes().front();
+    return (SrcMgr::CharacteristicKind)(getData().front() & ~IsSubmoduleBit);
   }
 
-  size_t getNumIncludes() const { return getNumReferences() - 1; }
+  bool isSubmodule() const { return (getData().front() & IsSubmoduleBit) != 0; }
+
+  std::optional<ObjectRef> getSubmoduleRef() const {
```

Comment:
👍 for the rename.  Submodule name is likely larger than an 8-byte internal ref, so I'd prefer to keep it simple in this case since IncludeTree already has a non-trivial data layout involving variable-sized data that we are managing manually.  If it turns out to be useful to change the layout we can always do that in the future.

Reply:
I'm more thinking of the hash size being bigger than the data itself, which is relevant for moving the graph to a different CAS, but I'm fine with keeping as-is for now.

Reply:
SGTM, happy to re-evaluate later if we find cases where the tradeoff is wrong.

Renamed to `getSubmoduleNameRef` in latest push.